### PR TITLE
feat: add enable_simple_view to PipelineJob.list()

### PR DESCRIPTION
**Pull request by sararob**
Add support for `read_mask` option in `PipelineJob.list()` to improve performance. This will reduce the number of fields returned in the response for each PipelineJob.

### CODE REVIEW

File: google/cloud/aiplatform/pipeline_jobs.py [hunk 1]
```diff
@@ -538,9 +544,34 @@ def list(
...
+                    "start_time",
+                    "end_time",
+                    "update_time",
+                    "labels",
+                    "template_uri",
+                    "template_metadata.version",
+                    "job_detail.pipeline_run_context",
+                    "job_detail.pipeline_context",
+                ]
+            )
```
[the remark this answers is not in the document]
Comment (sararob) [reply google/cloud/aiplatform/pipeline_jobs.py]: Done

File: google/cloud/aiplatform/base.py [hunk 2]
```diff
@@ -1026,6 +1027,7 @@ def _list(
         cls_filter: Callable[[proto.Message], bool] = lambda _: True,
         filter: Optional[str] = None,
         order_by: Optional[str] = None,
+        read_mask: Optional[field_mask.FieldMask] = None,
```
Comment (jaycee-li) [inline google/cloud/aiplatform/base.py:1030]: Could you please add doc string for this and the one in the next method?

File: google/cloud/aiplatform/pipeline_jobs.py [hunk 3]
```diff
@@ -524,6 +528,10 @@ def list(
                 Optional. A comma-separated list of fields to order by, sorted in
                 ascending order. Use "desc" after a field name for descending.
                 Supported fields: `display_name`, `create_time`, `update_time`
+            enable_simple_view (bool):
+                Optional. Whether to pass the `read_mask` parameter to the list call.
+                This will improve the performance of calling list(). However, the
+                returned PipelineJob list will not include all fields for each PipelineJob.
```
Comment (jaycee-li) [inline google/cloud/aiplatform/pipeline_jobs.py:534]: I think here we can specify what fields are missing.

**Review by jaycee-li — CHANGES_REQUESTED** [review]

**Review by jaycee-li — APPROVED** [review]

**Pull request merged**